code/datums/traits/_quirk.dm

```diff
@@ -12,6 +12,7 @@
 	var/mob_trait //if applicable, apply and remove this mob trait
 	///Amount of points this trait is worth towards the hardcore character mode; minus points implies a positive quirk, positive means its hard. This is used to pick the quirks assigned to a hardcore character. 0 means its not available to hardcore draws.
 	var/hardcore_value = 0
+	var/post_add_ran = FALSE /// Tracks if the post_add was run on the quirk on process tick
```

Comment:
Could you turn this into a bitflag? Something like `var/quirk_flags = NONE`, `#define QUIRK_POST_ADD_RAN (1<<0)` and `if(!(quirk_flags & QUIRK_POST_ADD_RAN))`

Reply:
Making this change.